src/main/webapp/search/results.js

```diff
@@ -0,0 +1,218 @@
+import React from 'react'
+
+import gql from 'graphql-tag'
+import { useQuery } from '@apollo/react-hooks'
+
+import Grid from '@material-ui/core/Grid'
+import LinearProgress from '@material-ui/core/LinearProgress'
+import Typography from '@material-ui/core/Typography'
+import Tabs from '@material-ui/core/Tabs'
+import Tab from '@material-ui/core/Tab'
+import Table from '@material-ui/core/Table'
+import TableRow from '@material-ui/core/TableRow'
+import TableCell from '@material-ui/core/TableCell'
+import TableBody from '@material-ui/core/TableBody'
+import TablePagination from '@material-ui/core/TablePagination'
+import TextFieldsIcon from '@material-ui/icons/TextFields'
+import ImageIcon from '@material-ui/icons/Image'
+
+import useSearchRouting, { DetailsLink } from './search-routing'
+import Search from './search'
+import Thumbnail from '../thumbnail/thumbnail'
+
+const LoadingComponent = () => <LinearProgress />
+
+const metacardQuery = gql`
+  query TextQuery($filter: Json!, $settings: QuerySettingsInput) {
+    metacards(filterTree: $filter, settings: $settings) {
+      attributes {
+        id
+        description
+        title
+        modified
+        thumbnail
+      }
+      status {
+        hits
+        elapsed
+      }
+    }
+  }
+`
+
+const Paging = props => {
+  const { total, startIndex, pageSize, handleRoute } = props
+
+  const handleChangePage = (_, n) => {
+    const startIndex = n * pageSize + 1
+    handleRoute({ startIndex: startIndex })
+  }
+  const handleChangeRowsPerPage = e => {
+    const size = parseInt(e.target.value, 10)
+    handleRoute({ startIndex: 1, pageSize: size })
+  }
+
+  return (
+    <TablePagination
+      component="div"
+      rowsPerPageOptions={[15, 50, 100]}
+      colSpan={12}
+      count={total}
+      rowsPerPage={pageSize}
+      page={startIndex > pageSize ? (startIndex - 1) / pageSize : 0}
+      onChangePage={handleChangePage}
+      onChangeRowsPerPage={handleChangeRowsPerPage}
+    />
+  )
+}
+
+const GalleryThumbnail = ({ thumbnail }) => {
+  return thumbnail !== undefined ? <Thumbnail src={thumbnail} /> : null
+}
+
+const GalleryItem = props => {
+  const { result } = props
+
+  return (
+    <Grid item xs={3}>
+      <DetailsLink id={result.id}>
```

Comment:
It's not obvious that I have to click the browser's back button to get from Details View back to results.

Reply:
That is intentional.  This is an example of a hybrid style application that should look and feel like a natural web page.  Similarly Google and Amazon do not have back buttons.  The URL is always up to date too so you can always copy the URL and get back to the same application state.